### PR TITLE
Implemented #559

### PR DESCRIPTION
**Pull request by RoiEXLab**
#559

Example
screenshot 5

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/635)

### CODE REVIEW

File: src/games/strategy/triplea/ui/screen/UnitsDrawer.java [hunk 1]
```diff
@@ -77,6 +76,12 @@ public void draw(final Rectangle bounds, final GameData data, final Graphics2D g
     final Image img = m_uiContext.getUnitImageFactory().getImage(type, owner, data,
         m_damaged > 0 || m_bombingUnitDamage > 0, m_disabled);
     graphics.drawImage(img, m_placementPoint.x - bounds.x, m_placementPoint.y - bounds.y, null);
+    if(!type.getName().equals("aaGun") && !type.getName().equals("factory")){//TODO add every "static" non-movable unit type of every map.
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/screen/UnitsDrawer.java]: We should have a unit type matcher that can accomplish this.

Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/screen/UnitsDrawer.java]: I wonder if we really want to exclude AA Gun - they can move during non-combat. It can be pretty easy to have a couple that are next to each other and make it difficult to tell apart.

Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/screen/UnitsDrawer.java]: The code addition is a bit hard to follow. The `draw` method is already pretty bad. Could you see about cleaning this up, making it a bit easier to understand/read?

**Comment by DanVanAtta** [comment]
Could be a very useful update.  For UI changes like this, would you mind posting screenshots of before and after?
Second, I think we may want to have this as a configuration setting so it can be turned on and off by users. Defaulting to having it on would be okay, and then presumably there would be a menu setting for those that want to turn it off. Would be nice if we could find an on-map area where it would make sense to have controls for the map rendering display settings. Till then though the main game menu bar is where setting options are grouped.

File: src/games/strategy/triplea/ui/screen/UnitsDrawer.java (hunk 1 above)
Comment (RoiEXLab) [inline src/games/strategy/triplea/ui/screen/UnitsDrawer.java]: The aaGun belongs to the nation being placed in (e.g. aaGun in Germany belongs to Germany etc.)

**Comment by RoiEXLab** [comment]
> Could be a very useful update. For UI changes like this, would you mind posting screenshots of before and after?

I inserted a Screenshot..

> Second, I think we may want to have this as a configuration setting so it can be turned on and off by users. Defaulting to having it on would be okay, and then presumably there would be a menu setting for those that want to turn it off. Would be nice if we could find an on-map area where it would make sense to have controls for the map rendering display settings. Till then though the main game menu bar is where setting options are grouped.

Good Point! I'll do that.

**Comment by RoiEXLab** [comment]
> We should have a unit type matcher that can accomplish this.

Well I didn't find one... If you can tell me the class name I will add that... Dumping those String into a list for now...

File: src/games/strategy/triplea/ui/TripleAFrame.java [hunk 2]
```diff
@@ -557,6 +562,7 @@ public void stopGame() {
     m_showHistoryAction = null;
     m_showMapOnlyAction = null;
     m_showCommentLogAction = null;
+    m_showUnitNationAction = null;
```
Comment (RoiEXLab) [inline src/games/strategy/triplea/ui/TripleAFrame.java]: All components are set to null due to a bug in java 5 (1.5) Should I just remove since I don't think this bug iis still present?

**Comment by RoiEXLab** [comment]
Created a setting working similar to the Show Units Setting...
And as alwasy I renamed Variables with the m_ prefix...

File: src/games/strategy/triplea/ui/TripleAFrame.java (hunk 2 above)
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/TripleAFrame.java]: let's do it. I've come across this code half a dozen times and have been tempted to as well.

File: src/games/strategy/triplea/ui/TripleAFrame.java [hunk 3]
```diff
@@ -2322,6 +2332,18 @@ private void showCommentLog() {
       }
     }
   };
+
+  private AbstractAction m_showUnitNationAction = new AbstractAction(){
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/TripleAFrame.java]: We have a utility library that will make most of this code not necessary:
    `SwingAction.java` -- `public static AbstractAction of(final Runnable swingAction) {`
```
SwingAction.of( e -> {
   UnitsDrawer.setN...
   geMapPanel().resetMap();
});
```

File: src/games/strategy/triplea/ui/TripleaMenu.java [hunk 4]
```diff
@@ -437,6 +438,15 @@ private void addShowCommentLog(final JMenu parentMenu) {
     showCommentLog.setModel(frame.getShowCommentLogButtonModel());
     parentMenu.add(showCommentLog).setMnemonic(KeyEvent.VK_L);
   }
+  
+  /**
+   * @param parentMenu
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/TripleaMenu.java]: nitpick: empty javadoc does not add any value, just a maintenance cost. please remove or alternatively pls add a meaningful description of hte parameter.

File: src/games/strategy/triplea/ui/screen/UnitsDrawer.java [hunk 5]
```diff
@@ -24,86 +25,92 @@
...
-    m_unitType = unitType;
-    m_playerName = playerName;
-    m_placementPoint = placementPoint;
-    m_damaged = damaged;
-    m_bombingUnitDamage = bombingUnitDamage;
-    m_disabled = disabled;
-    m_overflow = overflow;
-    m_territoryName = territoryName;
-    m_uiContext = uiContext2;
+    this.count = count;
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/screen/UnitsDrawer.java]: liking the variable rename. Given we are jumping to a new major version, it is a rare time where it is 100% safe (serialization and reflection make simple variable renames a headache sometimes)

File: src/games/strategy/triplea/ui/screen/UnitsDrawer.java [hunk 6]
```diff
@@ -24,86 +25,92 @@
...
-      throw new IllegalStateException("Type not found:" + m_unitType);
+      throw new IllegalStateException("Type not found:" + unitType);
+    }
+    final PlayerID owner = data.getPlayerList().getPlayerID(playerName);
+    final Image img = uiContext.getUnitImageFactory().getImage(type, owner, data,
+        damaged > 0 || bombingUnitDamage > 0, disabled);
+    graphics.drawImage(img, placementPoint.x - bounds.x, placementPoint.y - bounds.y, null);
+    if(!STATIC_UNITS.contains(type.getName()) && drawUnitNation){
+      final Image flag = uiContext.getFlagImageFactory().getSmallFlag(owner);
+      final int xoffset = img.getWidth(null) - flag.getWidth(null);//If someone wants to put more afford in this feature he could add an algorithm calculation the right-bottom corner
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/screen/UnitsDrawer.java]: be more precise with the TODO comment if you could. It does not necessarily make a lot of sense.

File: src/games/strategy/triplea/ui/screen/UnitsDrawer.java [hunk 7]
```diff
@@ -24,86 +25,92 @@
...
   public String getPlayer() {
-    return m_playerName;
+    return playerName;
   }
 
   @Override
   public void draw(final Rectangle bounds, final GameData data, final Graphics2D graphics, final MapData mapData,
       final AffineTransform unscaled, final AffineTransform scaled) {
-    if (m_overflow) {
+    if (overflow) {
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/screen/UnitsDrawer.java]: relatively complex code, could you see if you can simplify it?

**Comment by DanVanAtta** [comment]
Screenshot looks pretty good. Is it possible to draw the flag first, then the unit, so the flag appears behind the units? Perhaps we may really want the flag in front, in case a unit icon is giant. I think it might be cleaner with the flag behind the unit. Also pls do test World at War with this feature. It has a number of territories with densely packed units.

File: src/games/strategy/triplea/ui/screen/UnitsDrawer.java (hunk 7 above)
Comment (RoiEXLab) [inline src/games/strategy/triplea/ui/screen/UnitsDrawer.java]: The problem is, that I don't know exactly what most of the code does ^^ but sure I'll try

**Comment by RoiEXLab** [comment]
You mean like in the axis and allies board game, so the unit stands on top? 
Of course... I just have to move the drawing code a few lines up

File: src/games/strategy/triplea/ui/screen/UnitsDrawer.java [hunk 8]
```diff
@@ -24,86 +25,102 @@
...
     }
-    final PlayerID owner = data.getPlayerList().getPlayerID(m_playerName);
-    final Image img = m_uiContext.getUnitImageFactory().getImage(type, owner, data,
-        m_damaged > 0 || m_bombingUnitDamage > 0, m_disabled);
-    graphics.drawImage(img, m_placementPoint.x - bounds.x, m_placementPoint.y - bounds.y, null);
     // more then 1 unit of this category
-    if (m_count != 1) {
+    if (count != 1) {
       final int stackSize = mapData.getDefaultUnitsStackSize();
       if (stackSize > 0) { // Display more units as a stack
```
Comment (RoiEXLab) [inline src/games/strategy/triplea/ui/screen/UnitsDrawer.java]: Is this Feature even used in a single Map?
Do you know a map, in which Units are stacked on top of each other?

**Comment by RoiEXLab** [comment]
Have a look at the new Screenshot...

**Comment by RoiEXLab** [comment]
BTW World at War seems to run fine - It's a bit hard to test since I get a "empty" screen shortly after the world loaded...

**Comment by DanVanAtta** [comment]
I like the first screenshot quite a bit with the flags smaller and off to the right. Would you be able to restore that?

WaW dies on linux/Mac. You need to create a copy of the American "marine.png" and create a second one called "Marine.png"
Fixing that on all maps is a priority..

**Comment by RoiEXLab** [comment]
You may notice in the second Screenshot, that there is an options menu which enables you to choose between off the first and the second 
I asked my father as a strategic player what he likes the most and he actually likes the second option the most... 
The question is, what should be selected by default?

**Comment by RoiEXLab** [comment]
and on Windows systems it is not possible to have to files like Marine.png and marine.png at the same time since windows file system is not case sensitive -> The map would probably make download our whatever problems on Windows... I'm not actually sure what will happen since maps are distributed as zip file...

File: src/games/strategy/triplea/ui/MapPanel.java [hunk 9]
```diff
@@ -128,25 +126,25 @@ public void run() {
         });
       }
     });
-    recreateTiles(data, m_uiContext);
-    m_uiContext.addActive(new Active() {
+    recreateTiles(data, this.uiContext);
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/MapPanel.java]: prefer to avoid using 'this' when it is not necessary.  The rest of the project is relatively consistent about that. Small point.

File: src/games/strategy/triplea/ui/MapPanel.java [hunk 10]
```diff
@@ -237,69 +235,69 @@ public void run() {
...
   private void notifyTerritorySelected(final Territory t, final MouseDetails me) {
-    for (final MapSelectionListener msl : m_mapSelectionListeners) {
+    for (final MapSelectionListener msl : mapSelectionListeners) {
       msl.territorySelected(t, me);
     }
   }
 
   private void notifyMouseMoved(final Territory t, final MouseDetails me) {
-    for (final MapSelectionListener msl : m_mapSelectionListeners) {
+    for (final MapSelectionListener msl : mapSelectionListeners) {
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/MapPanel.java]: If you can, the variable renaming is nice in PRs on their own.  Allows the review to focus on the functional changes, and does not hold up the renaming in case there is more to do.

File: src/games/strategy/triplea/ui/TripleaMenu.java [hunk 11]
```diff
@@ -410,10 +403,37 @@ public void actionPerformed(final ActionEvent e) {
     });
     menuGame.add(mapZoom).setMnemonic(KeyEvent.VK_Z);
   }
+  
+  private void addUnitNationDrawMenu(final JMenu parentMenu){
+    final JMenu unitSizeMenu = new JMenu();
+    unitSizeMenu.setMnemonic(KeyEvent.VK_N);
+    unitSizeMenu.setText("Show Unit Nation");
+    final ButtonGroup unitNationGroup = new ButtonGroup();
+    final JRadioButtonMenuItem nationOption0 = new JRadioButtonMenuItem("Hide Nation");
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/TripleaMenu.java]: Nit:  vertical distance could be minimized. IE: avoid declaring variables at top, declare them at the last possible moment so they are as close to the usage as you can.

File: src/games/strategy/triplea/ui/TripleaMenu.java [hunk 12]
```diff
@@ -410,10 +403,37 @@ public void actionPerformed(final ActionEvent e) {
...
+    final JRadioButtonMenuItem nationOption1 = new JRadioButtonMenuItem("Next to Unit");
+    final JRadioButtonMenuItem nationOption2 = new JRadioButtonMenuItem("Below Unit");
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/TripleaMenu.java]: DRY violation, 5 lines of code where only two things change. I suggest a function call similar to:
`createRadioButtonItem("Below Unit", 0);`

Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/TripleaMenu.java]: To be clear, the 5 lines are, one for the new RadioButtonMenu Item, then 4 more to do the action listener.

File: src/games/strategy/triplea/ui/TripleaMenu.java [hunk 13]
```diff
@@ -410,10 +403,37 @@ public void actionPerformed(final ActionEvent e) {
...
+    }));
+    nationOption1.addActionListener(SwingAction.of(e -> {
+      UnitsDrawer.setNationVisible((byte)1);
+      frame.getMapPanel().resetMap();
+    }));
+    nationOption2.addActionListener(SwingAction.of(e -> {
+      UnitsDrawer.setNationVisible((byte)2);
+      frame.getMapPanel().resetMap();
+    }));
+    nationOption2.setSelected(true);
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/TripleaMenu.java]: Here is where you would want to make the option 'sticky', and have the chosen option be retained between game restarts. The other menu settings should have some examples.

To be explicit, around here you would load the user selected option, defaulting to something if not set, then when changed you would update that setting.

File: src/games/strategy/triplea/ui/screen/UnitsDrawer.java [hunk 14]
```diff
@@ -24,86 +25,102 @@
...
+      final int xoffset = img.getWidth(null) / 2 - flag.getWidth(null) / 2;//centered flag in the middle
+      final int yoffset = img.getHeight(null) / 2 - flag.getHeight(null) / 4 - 5;//centered flag in the middle moved it 1/2 - 5 down 
+      graphics.drawImage(flag, (placementPoint.x - bounds.x) + xoffset, (placementPoint.y - bounds.y) + yoffset, null);
+    }
+    //This Method draws the unit Image
+    graphics.drawImage(img, placementPoint.x - bounds.x, placementPoint.y - bounds.y, null);
+    //If Mode 1 is selected the nation Flag gets drawn next to the Unit
+    if(!STATIC_UNITS.contains(type.getName()) && drawUnitNationMode == 1){
+      final Image flag = uiContext.getFlagImageFactory().getSmallFlag(owner);
+      final int xoffset = img.getWidth(null) - flag.getWidth(null);//If someone wants to put more effort in this, he could add an algorithm to calculate the real
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/screen/UnitsDrawer.java]: The TODO comment should explain what we are doing today that is "not" calculating the real right corner. Said another way, after reading the comment, my first response is, "okay, if we want to draw in the real right corner, but are not, what are we doing that is different?"

File: src/games/strategy/triplea/ui/screen/UnitsDrawer.java [hunk 15]
```diff
@@ -24,86 +25,102 @@
...
+  private final String playerName;
+  private final Point placementPoint;
+  private final int damaged;
+  private final int bombingUnitDamage;
+  private final boolean disabled;
+  private final boolean overflow;
+  private final String territoryName;
+  private final IUIContext uiContext;
+  private final static List<String> STATIC_UNITS = Arrays.asList("aaGun","factory");//TODO add every "static" non-movable unit type of every map.
+  private static byte drawUnitNationMode = 2;
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/screen/UnitsDrawer.java]: Please use an enum instead of bit mask for `drawUnitNatioMode`

Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/screen/UnitsDrawer.java]: I also recommend replacing 'UnitNation' with 'UnitFlag'

File: src/games/strategy/triplea/ui/screen/UnitsDrawer.java [hunk 16]
```diff
@@ -24,86 +25,102 @@
...
+    final UnitType type = data.getUnitTypeList().getUnitType(unitType);
     if (type == null) {
-      throw new IllegalStateException("Type not found:" + m_unitType);
+      throw new IllegalStateException("Type not found:" + unitType);
+    }
+    final PlayerID owner = data.getPlayerList().getPlayerID(playerName);
+    final Image img = uiContext.getUnitImageFactory().getImage(type, owner, data, damaged > 0 || bombingUnitDamage > 0, disabled);
+    //If Mode 2 is selected the nation Flag gets drawn below the Unit
+    if(!STATIC_UNITS.contains(type.getName()) && drawUnitNationMode == 2){
+      final Image flag = uiContext.getFlagImageFactory().getFlag(owner);
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/screen/UnitsDrawer.java]: One note, when  'drawUnitNationMode' is switched to an enum, these if statements can move to a switch statement, then with the 'if' statement bodies extracted to helper methods, should then look pretty clean.

File: src/games/strategy/triplea/ui/screen/UnitsDrawer.java (hunk 8 above)
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/screen/UnitsDrawer.java]: Not sure. If not, and you're inclined, please remove it in a  follow-up PR after we get this one merged.

File: src/games/strategy/triplea/ui/screen/UnitsDrawer.java [hunk 17]
```diff
@@ -166,10 +183,14 @@ public int getLevel() {
...
   public String toString() {
-    return "UnitsDrawer for " + m_count + " " + MyFormatter.pluralize(m_unitType) + " in  " + m_territoryName;
+    return "UnitsDrawer for " + count + " " + MyFormatter.pluralize(unitType) + " in  " + territoryName;
   }
 
   private boolean isDamageFromBombingDoneToUnitsInsteadOfTerritories(final GameData data) {
     return games.strategy.triplea.Properties.getDamageFromBombingDoneToUnitsInsteadOfTerritories(data);
   }
+  
+  public static void setNationVisible(byte i){
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/screen/UnitsDrawer.java]: The static state is a code smell. We should be getting a props object injected into this class, but that is beyond the scope of this PR.

Regardless, this method would be helped with a renaming, something like:  `setUnitFlagDrawMode( UnitFlagDrawMode drawMode )`

**Comment by DanVanAtta** [comment]
Looking pretty close. A couple items to be address @RoiEXLab

**Comment by DanVanAtta** [comment]
Ah, you must be having a different problem. On Linux/Mac the map blows up because of the bad casing, but it is okay on windows.

File: src/games/strategy/triplea/ui/MapPanel.java (hunk 9 above)
Comment (RoiEXLab) [inline src/games/strategy/triplea/ui/MapPanel.java]: Well inthis case it is neccessary... this variable should refer the uiContext of the class, not of the method

File: src/games/strategy/triplea/ui/MapPanel.java (hunk 10 above)
Comment (RoiEXLab) [inline src/games/strategy/triplea/ui/MapPanel.java]: I will eventually

File: src/games/strategy/triplea/ui/TripleaMenu.java (hunk 11 above)
Comment (RoiEXLab) [inline src/games/strategy/triplea/ui/TripleaMenu.java]: Done

File: src/games/strategy/triplea/ui/screen/UnitsDrawer.java (hunk 16 above)
Comment (RoiEXLab) [inline src/games/strategy/triplea/ui/screen/UnitsDrawer.java]: The problem is, that the rendering order must retain the same... Using a switch statement would probably double some of the code...

File: src/games/strategy/triplea/ui/TripleaMenu.java (hunk 13 above)
Comment (RoiEXLab) [inline src/games/strategy/triplea/ui/TripleaMenu.java]: How would I do that, what system is tripleA using?

**Comment by RoiEXLab** [comment]
So the only thing left is to let the option setting be the user preffered setting. If anyone tells me what to to in order to load and save a configuration I will do that.

File: src/games/strategy/triplea/ui/screen/UnitsDrawer.java [hunk 18]
```diff
@@ -36,7 +36,10 @@
   private final String territoryName;
   private final IUIContext uiContext;
   private final static List<String> STATIC_UNITS = Arrays.asList("aaGun","factory");//TODO add every "static" non-movable unit type of every map.
-  private static byte drawUnitNationMode = 2;
+  private static UnitFlag drawUnitNationMode = UnitFlag.NEXT_TO;
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/screen/UnitsDrawer.java]: I suggest `UnitFlagDrawMode` for an enum name, more precise.
Also please update the variable name so it matches the enum.

File: src/games/strategy/triplea/ui/TripleaMenu.java [hunk 19]
```diff
@@ -407,30 +407,21 @@ public void actionPerformed(final ActionEvent e) {
   private void addUnitNationDrawMenu(final JMenu parentMenu){
     final JMenu unitSizeMenu = new JMenu();
     unitSizeMenu.setMnemonic(KeyEvent.VK_N);
-    unitSizeMenu.setText("Show Unit Nation");
+    unitSizeMenu.setText("Set Flag Display Mode");
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/TripleaMenu.java]: The word "Set" is not needed in a menu item, it is already implied a bit. Can be shortened down to "Flag Display Mode"

**Comment by DanVanAtta** [comment]
For the setting, see:
https://github.com/triplea-game/triplea/blob/master/src/games/strategy/performance/Perf.java
https://github.com/triplea-game/triplea/blob/master/src/games/strategy/performance/EnablePerformanceLoggingCheckBox.java

**Comment by DanVanAtta** [comment]
Also left a few comments, github marked them as out-dated even though they are fresh, pls double check that you hit them all.

**Comment by RoiEXLab** [comment]
Everthing works now fine as expected...

File: src/games/strategy/triplea/ui/screen/UnitsDrawer.java (hunk 16 above)
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/screen/UnitsDrawer.java]: Yeah, I perhaps should have mentioned the duplication. IMO it is not too bad since it is one liners, and it makes the ordering switch very clear, for example:
```
switch( .. ) { 
  case drawBefore:
       drawLargeUnitFlagImage();
       graphics.drawImage(img, placementPoint.x - bounds.x, placementPoint.y - bounds.y, null);;
 case drawAfter:
       graphics.drawImage(img, placementPoint.x - bounds.x, placementPoint.y - bounds.y, null);;
       drawSmallUnitFlagImage();
}  
```
IMO the extra line of duplicated code is nicer here compared to logic nested within conditionals. The conditionals require some extra effort:
- need to understand what the 4 lines of code are doing in each conditional, how they are different, and figure out why exactly that is more important. Hiding that info behind a helper function and switch statement does not necessarily make it simpler, but it does accomplish some information hiding, so a person can figure out the overall flow very easily, and can dig into the details if they want to (rather than just being forced to dig into the details).

Regardless of using the switch statement, the logic within the `if` statements would do well to be extracted to helper functions, makes the differences between the two cleaner, and allows for this method to become slightly simpler.

**Comment by RoiEXLab** [comment]
Now we are using a switch statement...

File: src/games/strategy/triplea/ui/TripleaMenu.java [hunk 20]
```diff
@@ -407,21 +408,25 @@ public void actionPerformed(final ActionEvent e) {
...
+    }
+    if(UnitsDrawer.UnitFlagDrawMode.NEXT_TO.toString().equals(setting)){
+      UnitsDrawer.setUnitFlagDrawMode(UnitsDrawer.UnitFlagDrawMode.NEXT_TO, prefs);    
+    }
+    if(UnitsDrawer.UnitFlagDrawMode.BELOW.toString().equals(setting)){
+      UnitsDrawer.setUnitFlagDrawMode(UnitsDrawer.UnitFlagDrawMode.BELOW, prefs);
+    }
+    
+    final ButtonGroup unitFlagSettingGroup = new ButtonGroup();
+    unitSizeMenu.add(createFlagDrawModeRadionButtonItem("Hide Nation", unitFlagSettingGroup, UnitsDrawer.UnitFlagDrawMode.NONE, setting, prefs));
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/TripleaMenu.java]: I recommend for menu option names: "None" "Small" "Large"
As is, they do not seem very consistent, and "Hide Nation" sounds a bit like all units would be hidden. 
The text is user facing, so kindly spending a bit of extra time on it to get it right is valuable.

**Comment by DanVanAtta** [comment]
@RoiEXLab  - please see comment about menu names. Let's follow up with that in a next PR and think of good labels. I think this is ready to be merged/integrated. Thank you for the good work.
